ui/components/app/currency-input/hooks/useTokenExchangeRate.tsx

```diff
@@ -22,18 +35,37 @@ export default function useTokenExchangeRate(
     shallowEqual,
   );
 
+  const [exchangeRates, setExchangeRates] = useState<Record<string, number>>(
+    {},
+  );
+
   return useMemo(() => {
     if (!tokenAddress) {
       return nativeConversionRate;
     }
 
     const contractExchangeRate =
-      contractExchangeRates[toChecksumAddress(tokenAddress)];
+      contractExchangeRates[tokenAddress] || exchangeRates[tokenAddress];
 
     if (!contractExchangeRate) {
+      fetchTokenExchangeRates(nativeCurrency, [tokenAddress], chainId).then(
+        (addressToExchangeRate) => {
+          setExchangeRates((prev) => ({
+            ...prev,
+            ...addressToExchangeRate,
+          }));
+        },
+      );
       return undefined;
     }
```

Comment:
How does the `contractExchangeRates` state get updated? Is it possible to dispatch that action instead and read contractExchangeRates once it's updated?

Reply:
I honestly couldn't figure out. These exchange rates are scoped for conversion rates so this seemed okay since it's not a ton of calls.